scipy/optimize/tests/test_bracket.py

```diff
@@ -855,7 +855,7 @@ def f(x):
 
         result = _bracket_minimum(f, xp.asarray(0.5535723499480897), xmin=xmin,
                                   xmax=xmax)
-        assert xmin == result.xl
+        xp_assert_close(result.xl, xmin)
```

Comment:
At least this should be true, otherwise there is no reason for the function to have terminated. Can we include this?
```suggestion
        xp_assert_close(result.xl, xmin)
        assert result.fl == f(xmin)
```

Reply:
This also fails, similar to https://github.com/scipy/scipy/pull/22216#discussion_r1900162288